orc8r/cloud/go/services/bootstrapper/servicers/registration/cloud_registration.go

```diff
@@ -0,0 +1,149 @@
+package registration
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"time"
+
+	"magma/orc8r/lib/go/protos"
+
+	"github.com/golang/glog"
+	"github.com/golang/protobuf/ptypes/timestamp"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+const (
+	nonceLength = 12
+	timeoutDuration = 30 * time.Minute
+	maxRetriesToGenerateNonce = 10
+)
+
+type cloudRegistrationServicer struct {
+	store Store
+}
+
+func NewCloudRegistrationServicer(store Store) (protos.CloudRegistrationServer, error) {
+	if store == nil {
+		return nil, fmt.Errorf("Storage store is nil")
+	}
+	return &cloudRegistrationServicer{store}, nil
+}
+
+func (crs *cloudRegistrationServicer) GetToken(c context.Context, request *protos.GetTokenRequest) (*protos.GetTokenResponse, error) {
+	n := request.Gateway.NetworkId
+	l := request.Gateway.LogicalId.Id
+
+	tokenInfo, err := crs.store.GetTokenInfoFromLogicalID(n, l)
+	if err != nil {
+		glog.Infof("Could not get tokenInfo for networkID %v and logicalID %v: %v", n, l, err)
+	}
+
+	refresh := request.Refresh || tokenInfo == nil || tokenTimedOut(tokenInfo)
+	if refresh {
```

Comment:
I didn't want to move it here since `tokenTimedOut` is used in `Registration` servicer as well, so I wanted to pull out the implementation?

Ok will add behavior explanation into proto definition!